examples/AdvancedOperations/AddDynamicPageFeed.php

```diff
@@ -0,0 +1,485 @@
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID,
+                $options[ArgumentNames::CAMPAIGN_ID] ?: self::CAMPAIGN_ID,
+                $options[ArgumentNames::AD_GROUP_ID] ?: self::AD_GROUP_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     * @param int $campaignId the campaign ID
+     * @param int $adGroupId the ad group ID
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        $customerId,
+        $campaignId,
+        $adGroupId
+    ) {
+        $dsaPageUrlLabel = "discounts";
+
+        // Gets the page feed details. This code example creates a new feed, but you can
+        // fetch and re-use an existing feed.
+        $feedDetails = self::createFeed($googleAdsClient, $customerId);
+        self::createFeedMapping($googleAdsClient, $customerId, $feedDetails);
+        self::createFeedItems($googleAdsClient, $customerId, $feedDetails, $dsaPageUrlLabel);
+
+        // Associates the page feed with the campaign.
+        self::updateCampaignDsaSetting($googleAdsClient, $customerId, $feedDetails, $campaignId);
+
+        // Optional: Targets web pages matching the feed's label in the ad group.
+        self::addDsaTarget($googleAdsClient, $customerId, $adGroupId, $dsaPageUrlLabel);
+
+        printf("Dynamic page feed setup is complete for campaign ID %s.%s", campaignId, PHP_EOL);
+    }
+
+    /**
+    * Creates a feed.
+    *
+    * @param GoogleAdsClient googleAdsClient the Google Ads API client.
```

Comment:
Some param/return descriptions have a period at the end and some not. It would be great to make it consistent in the entire file (on my side I never put any).

Reply:
Done.